MFVNeutralino/plugins/Vertexer.cc

```diff
@@ -1,891 +1,1771 @@
+            printf(" sv1'ntrack: %u \n", sv1.nTracks());
+          }
+
+          // only consider a pair with at least 3 tracks per vertex
+          if (sv0.nTracks() > 2 && sv1.nTracks() > 2) {
+
+            std::pair<bool, std::vector<std::vector<size_t>>> sharedjet_tool = sharedjets(vtxidx0, vtxidx1, sv_match_tracktojet_which_jetidx, sv_match_tracktojet_which_trkidx);
+
+            // loop thru {1,1}+{1,n} nsharedjets and remove just one shared track from v0 if a |dPhi(v0,one shared track)| > pi/2
+            if (sharedjet_tool.first) {
+              if (verbose)
+                printf("start shj implementation to {1,1} and {1,n} \n");
+              std::vector<std::vector<size_t>> sv_lonesharedtrack_which_trkidx = sharedjet_tool.second;
+              std::vector<size_t> sv0_lonesharedtrack_which_trkidx = sv_lonesharedtrack_which_trkidx[0];
+              std::vector<size_t> sv1_lonesharedtrack_which_trkidx = sv_lonesharedtrack_which_trkidx[1];
+              if (verbose) {
+                printf("size of set of lone shared tracks per sv0: %lu \n", sv0_lonesharedtrack_which_trkidx.size());
+                printf("size of set of lone shared tracks per sv1: %lu \n", sv1_lonesharedtrack_which_trkidx.size());
+              }
+              for (size_t k = 0; k < sv0_lonesharedtrack_which_trkidx.size(); k++) {
+                track_vec tks_sv0 = sv_total_track_which_trk_vec[vtxidx0];
+                size_t idx = sv0_lonesharedtrack_which_trkidx[k];
+                h_resolve_shared_jets_lonetrkvtx_dphi->Fill(fabs(reco::deltaPhi(tks_sv0[idx]->phi(), phi0)));
+
+                // drop the lone track pointing backwards from the vertex direction!
+                if (fabs(reco::deltaPhi(tks_sv0[idx]->phi(), phi0)) > M_PI / 2) {
+                  eraseElement(sv_total_track_which_trkidx[vtxidx0], idx);
+                }
+              }
+              if (verbose) {
+                printf("sv0'idx: %lu with ntrack before: %u",vtxidx0, sv0.nTracks());
+                printf("sv1'idx: %lu with ntrack before: %u",vtxidx1, sv1.nTracks());
+              }
+
+              track_set  sv0_resolved_sharedtracks_trkset;
+              for (unsigned int i = 0; i < sv_total_track_which_trkidx[vtxidx0].size(); ++i) {
+                int idx = sv_total_track_which_trkidx[vtxidx0][i];
+                track_vec tks_sv0 = sv_total_track_which_trk_vec[vtxidx0];
+                sv0_resolved_sharedtracks_trkset.insert(tks_sv0[idx]);
+
+              }
+              std::vector<reco::TransientTrack> sv0_resolved_sharedtracks_ttks;
+              for (auto tk : sv0_resolved_sharedtracks_trkset)
+                sv0_resolved_sharedtracks_ttks.push_back(tt_builder->build(tk));
+
+              reco::Vertex sv0_resolved_sharedtracks;
+
+              for (const TransientVertex& tv : kv_reco_dropin(sv0_resolved_sharedtracks_ttks))
+                sv0_resolved_sharedtracks = reco::Vertex(tv);
+
+              sv0 = sv0_resolved_sharedtracks;

```

Comment:
A comment here might be useful to mention that we're updating sv0, and are about to go back through the sv1 loop (just to make it super clear), since we're nested pretty deeply here.

Reply:
Okay. I add comments to clarify your previous comments in the code as well. Done.